### PR TITLE
[xdoctest] reformat example code with google style in No.339-No.343

### PR DESCRIPTION
**Pull request by longranger2**
### PR types
Others

### PR changes
Others

### Description
修改如下文件的示例代码为新的格式，并通过 xdoctest 检查：
- python/paddle/fluid/io.py
- python/paddle/fluid/layer_helper_base.py 
- python/paddle/fluid/layers/control_flow.py 
- python/paddle/fluid/layers/learning_rate_scheduler.py
- python/paddle/fluid/layers/math_op_patch.py

其中：
- io.py没有涉及到sample code；
- python/paddle/fluid/layers/control_flow.py  已经被迁移到 python/paddle/static/nn/control_flow.py；
- learning_rate_scheduler.py已经被删除

@sunzhongkai588 @SigureMo @megemini
- https://github.com/PaddlePaddle/Paddle/issues/55629

<br class="Apple-interchange-newline">

### CODE REVIEW

File: python/paddle/fluid/layers/math_op_patch.py [hunk 1]
```diff
@@ -429,14 +435,15 @@ def dim(self):
...
+                >>> paddle.enable_static()
 
-                # create a static Variable
-                x = paddle.static.data(name='x', shape=[3, 2, 1])
-                # print the dimension of the Variable
-                print(x.dim)
+                >>> # create a static Variable
+                >>> x = paddle.static.data(name='x', shape=[3, 2, 1])
+                >>> # print the dimension of the Variable
+                >>> print(x.dim)
```
Comment (SigureMo) [inline python/paddle/fluid/layers/math_op_patch.py]: `x.dim()`

File: python/paddle/fluid/layers/math_op_patch.py [hunk 2]
```diff
@@ -408,14 +413,15 @@ def ndimension(self):
...
+                >>> paddle.enable_static()
 
-                # create a static Variable
-                x = paddle.static.data(name='x', shape=[3, 2, 1])
-                # print the dimension of the Variable
-                print(x.ndimension)
+                >>> # create a static Variable
+                >>> x = paddle.static.data(name='x', shape=[3, 2, 1])
+                >>> # print the dimension of the Variable
+                >>> print(x.ndimension)
```
Comment (SigureMo) [inline python/paddle/fluid/layers/math_op_patch.py]: `x.ndimension()`

File: python/paddle/fluid/layers/math_op_patch.py [hunk 3]
```diff
@@ -258,30 +258,34 @@ def astype(self, dtype):
...
-                import paddle
-                import paddle.fluid as fluid
-                paddle.enable_static()
-                startup_prog = fluid.Program()
-                main_prog = fluid.Program()
-                with fluid.program_guard(startup_prog, main_prog):
-                    original_variable = paddle.static.data(name = "new_variable", shape=[2,2], dtype='float32')
-                    new_variable = original_variable.astype('int64')
-                    print("new var's dtype is: {}".format(new_variable.dtype))
+                >>> import paddle
```
Comment (SigureMo) [inline python/paddle/fluid/layers/math_op_patch.py:262]: 上面空一行

**Review by SigureMo — COMMENTED** [review]
fluid 已经 rename，这个 PR 可以继续了～

**Comment by SigureMo** [comment]
image

emmmm 解决冲突出问题了吧

**Comment by longranger2** [comment]
> emmmm 解决冲突出问题了吧

是的哈，重提了一个，https://github.com/PaddlePaddle/Paddle/pull/57087
